test/test_jit_cuda_fuser.py

```diff
@@ -508,6 +537,50 @@ def test_binary_ops(self):
         for op in operations:
             self._binary_test_helper(op)
 
+    @unittest.skipIf(not RUN_CUDA, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING,
+                     "Requires fusion optimization pass to be effective")
+    def test_binary_bitwise(self):
+        def jit_or(x: torch.Tensor, y: torch.Tensor, z: torch.Tensor):
+            return (x & y) | z
+
+        def jit_xor(x: torch.Tensor, y: torch.Tensor, z: torch.Tensor):
+            return (x & y) ^ z
+
+        def jit_lshift(x: torch.Tensor, y: torch.Tensor, z: torch.Tensor):
+            return (x & y) << z
+
+        def jit_rshift(x: torch.Tensor, y: torch.Tensor, z: torch.Tensor):
+            return (x & y) >> z
+
+        for jit_func in [jit_or, jit_xor, jit_lshift, jit_rshift]:
+            x = torch.randn(4, 8, 32, 32, dtype=torch.float, device="cuda").mul(5).to(torch.long)
+            y = torch.randn(4, 8, 32, 32, dtype=torch.float, device="cuda").mul(5).to(torch.long)
+            z = torch.randn(4, 8, 32, 32, dtype=torch.float, device="cuda").mul(2).to(torch.long)
+
+            jitted = torch.jit.script(jit_func)
+            jit_o = jitted(x, y, z)
+            jit_o = jitted(x, y, z)
+            o = jit_func(x, y, z)
+            self.assertEqual(o, jit_o)
+            self.assertGraphContains(jitted.graph_for(x, y, z), FUSION_GUARD)
+
+        # We shouldn't need this redefinition of the function, but otherwise it won't recompile for a new type
+        def jit_or(x: torch.Tensor, y: torch.Tensor, z: torch.Tensor):
```

Comment:
Note to myself, recompilation should work with type guard. But it then becomes tricky to confirm that on python side, so it's ok that we have another definition of the graph.